packages/cactus-test-tooling/src/main/typescript/iroha/iroha-test-ledger.ts

```diff
@@ -0,0 +1,352 @@
+    const label = "iroha-test-ledger";
+    const level = options.logLevel || "INFO";
+    this.log = LoggerProvider.getOrCreate({ level, label });
+  }
+
+  public getContainer(): Container {
+    const fnTag = "IrohaTestLedger#getContainer()";
+    if (!this.container) {
+      throw new Error(`${fnTag} container not yet started by this instance.`);
+    } else {
+      return this.container;
+    }
+  }
+
+  public get imageFqn(): string {
+    return `${this.imageName}:${this.imageVersion}`;
+  }
+
+  public async getRpcToriiPortHost(): Promise<string> {
+    const ipAddress = "127.0.0.1";
+    const hostPort: number = await this.getRpcToriiPort();
+    return `http://${ipAddress}:${hostPort}`;
+  }
+
+  /**
+   * Output is based on the standard Iroha genesis.block contents.
+   *
+   * @see https://github.com/hyperledger/iroha/blob/main/example/genesis.block
+   */
+  public getGenesisAdminAccount(): string {
+    return "admin@test";
+  }
+
+  /**
+   * Output is based on the standard Iroha admin user public key.
+   *
+   * @see https://github.com/hyperledger/iroha/blob/main/example/admin%40test.pub
+   * @see https://github.com/hyperledger/iroha/blob/main/example/genesis.block
+   */
+  public getGenesisAccountPubKey(): string {
+    return "313a07e6384776ed95447710d15e59148473ccfc052a681317a72a69f2a49910";
+  }
+
+  /**
+   * Output is based on the standard Iroha admin user private key.
+   *
+   * @see https://github.com/hyperledger/iroha/blob/main/example/admin%40test.priv
+   */
+  public getGenesisAccountPrivKey(): string {
+    return "f101537e319568c765b2cc89698325604991dca57b9716b58016b253506cab70";
+  }
+
+  public async getNodeKeyPair(): Promise<IKeyPair> {
+    const fnTag = `IrohaTestLedger#getNodeKeyPair()`;
+    if (!this.container) {
+      throw new Error(`${fnTag} this.container cannot be falsy.`);
+    }
+    const publicKey = await Containers.pullFile(
+      this.container,
+      "/opt/iroha_data/node0.pub",
+    );
+    const privateKey = await Containers.pullFile(
+      this.container,
+      "/opt/iroha_data/node0.priv",
+    );
+    return { publicKey, privateKey };
+  }
+
+  public async start(omitPull = false): Promise<Container> {
+    if (this.container) {
+      await this.container.stop();
+      await this.container.remove();
+    }
+    const docker = new Docker();
+    if (!omitPull) {
+      this.log.debug(`Pulling container image ${this.imageFqn} ...`);
+      await Containers.pullImage(this.imageFqn);
+      this.log.debug(`Pulled ${this.imageFqn} OK. Starting container...`);
+    }
+
+    return new Promise<Container>((resolve, reject) => {
+      const eventEmitter: EventEmitter = docker.run(
+        this.imageFqn,
+        [],
+        [],
+        {
+          ExposedPorts: {
+            [`${this.rpcToriiPort}/tcp`]: {}, // Iroha RPC - Torii
+          },
+          Env: this.envVars,
+          Healthcheck: {
+            Test: ["CMD-SHELL", "python3 /healthcheck.py admin@test 50051"],
```

Comment:
1. Is the file in absolute directory `/`? I think it is better to use different one
2. I suggest to use `getGenesisAdminAccount()` instead of `admin@test`, the same with `50051`

Reply:
1. Yes. The file is in the absolute directory `/`. I put it here because Iroha team put their entrypoint.sh and wait-for-it.sh at the same absolute directory `/`.
2. Unfortunately, I do not think I can put any function or string literal at line 192. It won't compile.

Reply:
OK, thanks for explanation, so I'm making resolved

Reply:
You are welcome. :)

Reply:
@hxlaf Sorry I'm late for the review party!

> Unfortunately, I do not think I can put any function or string literal at line 192. It won't compile.

Let's talk more about this, I'm pretty sure we can make it work in a way that Greg was asking for it (I also agree with the request)
Can you post here what the compiler error is when you change the code so that the credentials and the port are coming from a variable instead of being magic strings?

Reply:
Sorry. I must admit that I made a dumb mistake! It works. Thank you for bringing up the issue again.
![image](https://user-images.githubusercontent.com/10954157/129647990-4441841b-33bf-40a2-be9c-342d18ef0a0b.png)
I confused this part with the HostConfig section. I said no because I tried numerous times to put a string literal in the HostConfig section but failed. I am doing a couple of more tests and will upload the fix once the tests complete.

Reply:
Done with testing and squashing! Thank you!